packages/core/src/lib/upload.ts

```diff
@@ -20,13 +19,12 @@ export async function upload(
   options: UploadOptions,
   uploadFn: typeof uploadToPortal = uploadToPortal,
 ) {
-  const persist = normalizePersistConfig(options.persist);
-  if (!options.upload) {
-    throw new Error('upload config must be set');
+  if (options.upload == null) {
```

Comment:
I get it's same logical outcome. But the value will never be null. It will me `undefined` if ever.

Is this a best practice I don't know?

Reply:
This is a common practice we've established at FlowUp. Not sure how widespread it is, but the fact the [ESLint core `eqeqeq` rule supports a `'always', { null: 'never' }` option](https://eslint.org/docs/latest/rules/eqeqeq#always) is some indication there's split over it in the comminuty :thinking: 

The rationale for `x == null` and `x != null` is to treat `null` and `undefined` the same way, because there's no consistent semantic distinction between them. Having two distinct nullish values is a very strange quirk of JavaScript. In practice, picking `null` or `undefined` is basically arbitrary, so checking for a nullish value with minimal syntax is a practical and defensive approach in general. The key question for me is: "If you expected `undefined` and received `null` instead, would you want to treat it any differently?"